Write a Timed File Reader application

gmkv opened the pull request:
`timedfilereader` reads a file of arbitrary size for a given period of time
in milliseconds.

        usage: timedfilereader [filename] [timeout-in-milliseconds]

- If the whole file has been read within the given time period,
  a message will be printed showing how many bytes have been read,
  and the program will exit with status **(1)**.
- If the file has only been read partly within the given time period,
  a message will be printed showing how many bytes have been read,
  and the program will exit with status **(-1)**.

mgenov commented on timedio/timedfilereader.go:
```diff
@@ -0,0 +1,89 @@
+
+}
+
+func TimedFileReader(fileName string, readWholeFile *bool, numBytesRead chan<- int, quit <-chan time.Time) {
```
Name doesn't sound good as it's not a verb.

Also the names of functions defined in main could be with internal visibility as they cannot be exported to someone else.

mgenov commented on timedio/timedfilereader.go:
```diff
@@ -0,0 +1,89 @@
+func TimedFileReader(fileName string, readWholeFile *bool, numBytesRead chan<- int, quit <-chan time.Time) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalf("Error when opening file %q: %v\n", fileName, err)
```
It would be better error to be propagated one level up, e.g to be returned.

gmkv commented:
I changed ```TimedFileReader``` to ```readFileTimeout``` to better illustrate its behaviour, and added another parameter to emit any errors which may occur during processing of the file, since functions in goroutines don't return any values.

gmkv commented:
Hi Miro, does this look okay after I have implemented the last change?

gmkv commented on timedio/timedfilereader.go:
```diff
@@ -0,0 +1,105 @@
+			}
+			return
+		default:
+			n, err := file.Read(buf)
```
So, in order to end a read operation which has blocked (running in a seperate goroutine), I should close the file in a different goroutine after the time runs out, which would cause the read operation (back in the first goroutine) to end?

mgenov replied:
Yes.

gmkv commented:
I have made the suggested change to prevent the case where the read operation might block indefinitely.
